SpatialGDK/Source/SpatialGDK/Private/Interop/WorkingSetsCommon.cpp

```diff
@@ -0,0 +1,200 @@
+
+				EntitiesToRefresh.Emplace(MemberEntity);
+			}
+		}
+		break;
+		case EntityDelta::REMOVE:
+			RemovedWorkingSets.Emplace(MarkerEntityId);
+			UE_LOG(LogSpatialWorkingSetsCommon, Display, TEXT("Working set lost. MarkerEntityId: %lld EntityCount: %d"), MarkerEntityId,
+				   DataStorage[Delta.EntityId].Response.ActualEntities.Num());
+
+			for (const Worker_EntityId MemberEntity : DataStorage[Delta.EntityId].Response.ActualEntities)
+			{
+				EntityWorkingSetMembership.Remove(MemberEntity);
+
+				EntitiesToRefresh.Emplace(MemberEntity);
+			}
+			DataStorage.Remove(Delta.EntityId);
+			break;
+		case EntityDelta::UPDATE:
+		{
+			UpdatedWorkingSets.Emplace(MarkerEntityId);
+			auto& AddedSet = DataStorage[Delta.EntityId];
+
+			const auto MembersBeforeUpdate = AddedSet.Response.ActualEntities;
+
+			AddedSet.Update(Delta);
+
+			const auto& MembersAfterUpdate = AddedSet.Response.ActualEntities;
+
+			const auto ReleasedEntities = MembersBeforeUpdate.Difference(MembersAfterUpdate);
+			for (const Worker_EntityId_Key ReleasedEntityId : ReleasedEntities)
+			{
+				EntityWorkingSetMembership.Remove(ReleasedEntityId);
+
+				EntitiesToRefresh.Emplace(ReleasedEntityId);
+			}
+
+			const auto NewMemberEntities = MembersAfterUpdate.Difference(MembersBeforeUpdate);
+			for (const Worker_EntityId_Key MemberEntity : NewMemberEntities)
+			{
+				EntityWorkingSetMembership.Emplace(MemberEntity, Delta.EntityId);
+
+				EntitiesToRefresh.Emplace(MemberEntity);
+			}
+		}
+		break;
+		}
+	}
+}
+
+TArray<ComponentData> CreateWorkingSetComponents(const Worker_PartitionId AuthServerWorkerPartitionId,
+												 const FWorkingSetMarkerRequest& Request)
+{
+	TArray<ComponentData> Components;
+
+	Components.Emplace(Request.CreateComponentData());
+	Components.Emplace(FWorkingSetMarkerResponse().CreateComponentData());
+
+	auto Convert = [](Worker_ComponentData Component) {
+		ComponentData Converted(OwningComponentDataPtr(Component.schema_type), Component.component_id);
+		return Converted;
+	};
+
+	{
+		Components.Emplace(Convert(Metadata(TEXT("WorkingSetMarker")).CreateComponentData()));
+	}
+
+	{
+		// Give the auth UnrealWorker authority over the Request component, and the strategy worker auth over the Response component
+		AuthorityDelegation Delegation;
+		Delegation.Delegations.Emplace(SpatialConstants::STRATEGY_WORKER_AUTH_COMPONENT_SET_ID,
+									   SpatialConstants::INITIAL_STRATEGY_PARTITION_ENTITY_ID);
+		Delegation.Delegations.Emplace(SpatialConstants::LB_DELEGATION_AUTH_COMPONENT_SET_ID,
+									   SpatialConstants::INITIAL_STRATEGY_PARTITION_ENTITY_ID);
+		Delegation.Delegations.Emplace(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID, AuthServerWorkerPartitionId);
+
+		Components.Emplace(Convert(Delegation.CreateComponentData()));
+	}
+	{
+		Interest In;
+		Query Query;
+		QueryConstraint SelfConstraint;
+		SelfConstraint.bSelfConstraint = true;
+		Query.Constraint = SelfConstraint;
+		Query.ResultComponentIds = {
```

Comment:
Split that interest into two - one for ServerWorkers, another for the StrategyWorker.